pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -363,6 +363,7 @@ func NewVitessShardStatus() VitessShardStatus {
 		HasMaster:        corev1.ConditionUnknown,
 		HasInitialBackup: corev1.ConditionUnknown,
 		Idle:             corev1.ConditionUnknown,
+		Conditions:       make(map[VitessShardConditionType]*VitessShardCondition),
```

Comment:
This makes sense at a conceptual level (`NewX()` should allocate the map for you), but it actually doesn't have any practical effect right now. We currently only call `NewVitessShardStatus()` in one place, and then immediately overwrite `Conditions` with something else.

Reply:
Oh hmm, that's true I suppose. What do you suggest then as a better approach?